FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_light_weight/DemoTasks/LightWeightMQTTExample.c

```diff
@@ -41,41 +54,938 @@
+        LogInfo( ( "Attempt to subscribed to the MQTT topic %s.\r\n", mqttexampleTOPIC ) );
+        prvMQTTSubscribeToTopic( xMQTTSocket );
+
+        /* Process incoming packet from the broker. After sending the subscribe, the
+         * client may receive a publish before it receives a subscribe ack. Therefore,
+         * call generic incoming packet processing function. Since this demo is
+         * subscribing to the topic to which no one is publishing, probability of
+         * receiving Publish message before subscribe ack is zero; but application
+         * must be ready to receive any packet.  This demo uses the generic packet
+         * processing function everywhere to highlight this fact. */
+        prvMQTTProcessIncomingPacket( xMQTTSocket );
+
+        /**************************** Publish and Keep Alive Loop. ******************************/
+        /* Publish messages with QOS0, send and process Keep alive messages. */
+        for( ulPublishCount = 0; ulPublishCount < ulMaxPublishCount; ulPublishCount++ )
+        {
+            LogInfo( ( "Publish to the MQTT topic %s.\r\n", mqttexampleTOPIC ) );
+            prvMQTTPublishToTopic( xMQTTSocket );
+
+            /* Process incoming publish echo, since application subscribed to the same
+             * topic the broker will send publish message back to the application. */
+            LogInfo( ( "Attempt to receive publish message from broker.\r\n" ) );
+            prvMQTTProcessIncomingPacket( xMQTTSocket );
+
+            /* Leave Connection Idle for some time */
+            LogInfo( ( "Keeping Connection Idle.\r\n\r\n" ) );
+            vTaskDelay( mqttexampleKEEP_ALIVE_DELAY );
+
+            /* Send Ping request to broker and receive ping response */
+            LogInfo( ( "Sending Ping Request to the broker.\r\n" ) );
+            prvMQTTKeepAlive( xMQTTSocket );
+
+            /* Process Incoming packet from the broker */
+            prvMQTTProcessIncomingPacket( xMQTTSocket );
+        }
+
+        /************************ Unsubscribe from the topic. **************************/
+        LogInfo( ( "Unsubscribe from the MQTT topic %s.\r\n", mqttexampleTOPIC ) );
+        prvMQTTUnsubscribeFromTopic( xMQTTSocket );
+
+        /* Process Incoming packet from the broker. */
+        prvMQTTProcessIncomingPacket( xMQTTSocket );
+
+        /**************************** Disconnect. ******************************/
+
+        /* Send an MQTT Disconnect packet over the already connected TCP socket.
+         * There is no corresponding response for the disconnect packet. After sending
+         * disconnect, client must close the network connection. */
+        LogInfo( ( "Disconnecting the MQTT connection with %s.\r\n", democonfigMQTT_BROKER_ENDPOINT ) );
+        prvMQTTDisconnect( xMQTTSocket );
+
+        /* Close the network connection.  */
+        prvGracefulShutDown( xMQTTSocket );
+
+        /* Wait for some time between two iterations to ensure that we do not
```

Comment:
This is another reason not to use a public service that you do not operate here.  See my previous comment, this demo should use a local broker only.